das-jdbc/src/main/java/org/wso2/das/jdbcdriver/common/ServiceUtil.java

```diff
@@ -0,0 +1,202 @@
+        } else if (literal instanceof BigDecimal) {
+            retval = ServiceConstants.DATATYPES.DATATYPE_BIGDECIMAL;
+        } else if (literal instanceof Date) {
+            retval = ServiceConstants.DATATYPES.DATATYPE_DATE;
+        } else if (literal instanceof Time) {
+            retval = ServiceConstants.DATATYPES.DATATYPE_TIME;
+        } else if (literal instanceof Timestamp) {
+            retval = ServiceConstants.DATATYPES.DATATYPE_TIMESTAMP;
+        } else if (literal instanceof InputStream) {
+            retval = ServiceConstants.DATATYPES.DATATYPE_ASCIISTREAM;
+        }
+        return retval;
+    }
+
+    /**
+     * Match Regular expression with the given input string.
+     *
+     * @param likePattern Pattern to be matched
+     * @param escape      Escape sequence
+     * @param input       Input string which needs to match with the given pattern
+     */
+    public static boolean isPatternMatched(String likePattern, String escape, CharSequence input) {
+        boolean retval;
+        int percentIndex = likePattern.indexOf('%');
+        int underscoreIndex = likePattern.indexOf('_');
+        if (percentIndex < 0 && underscoreIndex < 0) {
+            retval = likePattern.equals(input); //No wildcards. Compare the string.
+        } else {
+            boolean isEscaped = false;
+            StringBuilder regex = new StringBuilder();
+            StringTokenizer tokenizer = new StringTokenizer(likePattern, "%_" + escape, true);
+            while (tokenizer.hasMoreTokens()) {
+                String token = tokenizer.nextToken();
+                if (token.equals(escape)) {
+                    if (isEscaped) {
+                        regex.append(Pattern.quote(token));
+                    } else {
+                        isEscaped = true;
+                    }
+                } else {
+                    if (isEscaped) {
+                        regex.append(Pattern.quote(token));
+                    } else if (token.equals("%")) {
+                        regex.append(".*");
+                    } else if (token.equals("_")) {
+                        regex.append(".");
+                    } else {
+                        regex.append(Pattern.quote(token));
+                    }
+                    isEscaped = false;
+                }
+            }
+            Pattern pattern = Pattern.compile(regex.toString());
+            retval = pattern.matcher(input).matches();
+        }
+        return retval;
+    }
+
+    /**
+     * Get the byte array of a given string.
+     *
+     * @param str Input string which needs to convert to byte[]
+     */
+    public static byte[] parseBytes(String str) {
+        try {
+            byte[] b;
+            if (str == null) {
+                b = null;
+            } else {
+                b = str.getBytes();
+            }
+            return b;
+        } catch (RuntimeException e) {
+            return null;
+        }
+    }
```

Comment:
This method can be simplified, which will avoid the explicit initialization of the byte array as well as catching the unchecked exception (unless of course we want to prevent the program halting at this stage):
```
if (str != null) {
    return str.getBytes();
}
return null;
```